### PR TITLE
Re-implement Android Editbox.

### PR DESCRIPTION
**Pull request by zilongshanren**
Now the EditBox behavior on Android platform is the same as iOS, maybe even better than iOS.
It supports Multiline input.  
More tests are needed to be done before the PR could be merged.

### CODE REVIEW

**Comment by zilongshanren** [comment]
It also fixes a bug in https://github.com/cocos2d/cocos2d-x/issues/9572

**Comment by zilongshanren** [comment]
it fixes issue https://github.com/cocos2d/cocos2d-x/issues/13147

File: cocos/ui/UIEditBox/UIEditBoxImpl-winrt.cpp [hunk 1]
```diff
@@ -392,14 +392,27 @@ bool UIEditBoxImplWinrt::initWithSize( const Size& size )
...
-    if(m_pLabel != NULL) {
-        m_pLabel->setSystemFontName(pFontName);
-        m_pLabel->setSystemFontSize(fontSize);
+    if(m_pLabel != NULL)
+    {
+        if(strlen(pFontName) > 0)
+        {
+            m_pLabel->setSystemFontName(pFontName);
+        }
+        if(fontSize != -1)
```
Comment (zilongshanren) [inline cocos/ui/UIEditBox/UIEditBoxImpl-winrt.cpp]: maybe I need to change it to `fontSize` > 0

**Comment by zilongshanren** [comment]
@WenhaiLin  @pandamicro  please help to review this PR.

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java [hunk 2]
```diff
@@ -268,6 +319,17 @@ protected void onCreate(final Bundle savedInstanceState) {
...
             mWebViewHelper = new Cocos2dxWebViewHelper(mFrameLayout);
         }
+
+        if(mEditBoxHelper == null){
+            mEditBoxHelper = new Cocos2dxEditBoxHelper(mFrameLayout);
+        }
+
+        Window window = this.getWindow();
+        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
+//        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
```
Comment (WenhaiLin) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java]: Please remove unnecessary code.

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java [hunk 3]
```diff
@@ -378,7 +436,11 @@ public void init() {
 
         // Set framelayout as the content view
         setContentView(mFrameLayout);
+
```
Comment (WenhaiLin) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java]: Please remove unnecessary blank line.

File: cocos/platform/android/jni/Java_org_cocos2dx_lib_Cocos2dxHelper.cpp [hunk 4]
```diff
@@ -104,33 +106,34 @@ void showDialogJNI(const char * message, const char * title) {
     }
 }
 
-void showEditTextDialogJNI(const char* title, const char* message, int inputMode, int inputFlag, int returnType, int maxLength, EditTextCallback callback, void* ctx) {
-    if (message == nullptr) {
-        return;
-    }
+// void showEditTextDialogJNI(const char* title, const char* message, int inputMode, int inputFlag, int returnType, int maxLength, EditTextCallback callback, void* ctx) {
```
Comment (WenhaiLin) [inline cocos/platform/android/jni/Java_org_cocos2dx_lib_Cocos2dxHelper.cpp]: Please remove unused code

Comment (zilongshanren) [inline cocos/platform/android/jni/Java_org_cocos2dx_lib_Cocos2dxHelper.cpp]: TODO: This commented code should also be removed.

File: cocos/ui/UIEditBox/UIEditBoxImpl-android.cpp [hunk 5]
```diff
@@ -28,18 +28,133 @@
...
+        env->ReleaseStringUTFChars(text, charText);
+        Director *dir = Director::getInstance();
+        Scheduler *sched = dir->getScheduler();
+        sched->performFunctionInCocosThread( std::bind(editBoxEditingDidChanged,index, textString));
+        // editBoxEditingDidChanged(index, textString);
+    }
+
+    JNIEXPORT void JNICALL Java_org_cocos2dx_lib_Cocos2dxEditBoxHelper_editBoxEditingDidEnd(JNIEnv *env, jclass, jint index, jstring text) {
+        // editBoxEditingDidEnd(index);
+        auto charText = env->GetStringUTFChars(text, NULL);
```
Comment (WenhaiLin) [inline cocos/ui/UIEditBox/UIEditBoxImpl-android.cpp:63]: I suggest you replace `jstring` with `jbyteArray` .For more information: https://github.com/cocos2d/cocos2d-x/issues/9150

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java [hunk 6]
```diff
@@ -39,14 +32,67 @@ of this software and associated documentation files (the "Software"), to deal
...
+import org.cocos2dx.lib.Cocos2dxHelper.Cocos2dxHelperListener;
+
+import javax.microedition.khronos.egl.EGL10;
+import javax.microedition.khronos.egl.EGLConfig;
+import javax.microedition.khronos.egl.EGLDisplay;
+
+class ResizeLayout extends FrameLayout{
+    private  boolean mEnableForceDoLayout = false;
+
+    public interface OnResizeListener {
```
Comment (WenhaiLin) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java]: Please remove unused `interface`.

**Pull request closed without merging**